include/internal/quic_types.h

```diff
@@ -18,17 +18,22 @@
 # ifndef OPENSSL_NO_QUIC
 
 /* QUIC encryption levels. */
-#  define QUIC_ENC_LEVEL_INITIAL          0
-#  define QUIC_ENC_LEVEL_HANDSHAKE        1
-#  define QUIC_ENC_LEVEL_0RTT             2
-#  define QUIC_ENC_LEVEL_1RTT             3
-#  define QUIC_ENC_LEVEL_NUM              4
+enum {
+    QUIC_ENC_LEVEL_INITIAL,
+    QUIC_ENC_LEVEL_HANDSHAKE,
+    QUIC_ENC_LEVEL_0RTT,
+    QUIC_ENC_LEVEL_1RTT,
+    QUIC_ENC_LEVEL_NUM       /* Must be the ultimate entry */
+};
 
 /* QUIC packet number spaces. */
-#  define QUIC_PN_SPACE_INITIAL           0
-#  define QUIC_PN_SPACE_HANDSHAKE         1
-#  define QUIC_PN_SPACE_APP               2     /* Must be last */
-#  define QUIC_PN_SPACE_NUM               3
+enum {
+    QUIC_PN_SPACE_INITIAL,
```

Comment:
I know the default starting number for enums is `0`, but I personally like to see it explicit. However, if it's standard not to use it, that's fine too

Reply:
I don't think the actual values matter here, so leaving it out seemed reasonable.

I don't think our style guide says anything about explicit starts for enum values.  We've rarely used them in the past due to portability issues passing them as arguments.

Reply:
> I don't think the actual values matter here

If `_NUM` is to be the last, and have the correct value, the first must be `0`.

My preference is to have the explicit (though redundant) `= 0` on the first one, but that's "just me" :)

Reply:
I don't know what I was think....
Fried brain again.

I've added the = 0.

Reply:
My approval still stands - thanks